antarest/study/service.py

```diff
@@ -213,17 +228,37 @@ def _generate_timeseries(self) -> None:
                 )
             )
 
-    def run_task(self, notifier: TaskUpdateNotifier) -> TaskResult:
+    def run_task(self, notifier: TaskUpdateNotifier, listener: t.Optional[ICommandListener]) -> TaskResult:
```

Comment:
Se here we should receive still only the task notifier / listener, and in the method we create the command listener implementation which will forward progress to the task listener.